core/src/main/java/org/apache/carbondata/core/indexstore/SegmentPropertiesFetcher.java

```diff
@@ -34,7 +35,7 @@
    * @return
    * @throws IOException
    */
-  SegmentProperties getSegmentProperties(Segment segment)
+  SegmentProperties getSegmentProperties(Segment segment, List<PartitionSpec> partitionSpecs)
```

Comment:
suggest to add one more interface instead of modifying existing one, keep both interface so that user can choose to use original one if no partition pruning is required.
and please modify function description accordingly

Reply:
done